doc/crypto/api.db/psa/crypto.h

```diff
@@ -505,6 +509,11 @@ psa_status_t psa_export_public_key(psa_key_id_t key,
                                    size_t * data_length);
 psa_status_t psa_generate_key(const psa_key_attributes_t * attributes,
                               psa_key_id_t * key);
+psa_status_t psa_generate_key_ext(const psa_key_attributes_t *attributes,
+                                  const psa_key_production_parameters_t *params,
+                                  const uint8_t *params_data,
+                                  size_t params_data_length,
+                                  mbedtls_svc_key_id_t *key);
```

Comment:
In Mbed TLS 3.6.1, we want to keep the existing beta `psa_generate_key_ext` and `psa_key_derivation_output_key_ext` with a `psa_key_production_parameters_t*` parameter with a flexible array member, for backward compatibility, and also provide the new C++-compatible type and function with a different name. So we'd like the revised function and type names to be different from the ones in the beta. I propose to go back to “custom”:

```
psa_status_t psa_generate_key_custom(const psa_key_attributes_t *attributes,
                                     const psa_key_custom_production_t *custom,
                                     const uint8_t *custom_data,
                                     size_t custom_data_length,
                                     mbedtls_svc_key_id_t *key);
```

Reply:
For the type, `psa_key_custom_production_t` or `psa_custom_key_production_t`? Both work. It's a custom key production (a key production that's custom), not a custom-key production (a production of a custom key). But `custom_key_production` is naturally understood as “custom-key production” which gives the wrong idea. So we lean towards `psa_key_custom_production_t`, which is less discoverable but less ambiguous.